lib/openapi.js

```diff
@@ -0,0 +1,262 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const yaml = require('js-yaml')
+const { formatParamUrl, plainJsonObjectToSwagger2, swagger2ParametersToOpenapi3, stripBasePathByServers } = require('./util')
+
+module.exports = function (opts, routes, Ref, cache, next) {
```

Comment:
this is a huge method that does many things. Would you be open to break it into several smaller functions, potentially spanning multiple files, to make it easier to follow the flow?

Reply:
I think it is hard for the other to follow the flow when it because smaller components. The flow here is very strict forward. Maybe extract all the inline function will make it better.

Reply:
It's less about branches and more about the difficulty of grasping many elements at once. Human brain is not really wired for that.
See https://en.wikipedia.org/wiki/The_Magical_Number_Seven,_Plus_or_Minus_Two
and https://www.informit.com/articles/article.aspx?p=1398607 for more details on this topic.

Reply:
This maybe a little bit out of scope. For your provided link, we need a lot of refactor in order to achieve it, but this PR is mainly for adding the openapi 3 support.
I do the refactor first because most of the functions are commonly use between ```openapi``` and ```swagger``` but it should not be the goal in this PR.

Reply:
I will left a comment to address the issue.